src/api/informer.rs

```diff
@@ -131,24 +133,25 @@ where
 
         // First check if we need to resync, if so reset our resource version
         // and wait a bit before proceeding.
-        // We take a read only lock here as most of the time it will not be necessary
-        // to take an exclusive lock.
-        if *self.needs_resync.read().unwrap() {
+        let mut needs_resync = self.needs_resync.lock().await;
+        if *needs_resync {
             // If desynched due to mismatching resourceVersion, retry in a bit
             let dur = Duration::from_secs(10);
             Delay::new(dur).await;
             self.reset().await?;
-            *self.needs_resync.write().unwrap() = false;
+            *needs_resync = false;
```

Comment:
you're holding this lock for 10 seconds here - but I guess that's fine. we're only meant to have one driving thread/context.